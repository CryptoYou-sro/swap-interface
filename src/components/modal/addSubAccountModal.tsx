import { useState } from 'react';
import { toast } from 'react-toastify';
import styled, { css } from 'styled-components';
import { BASE_URL, useStore } from '../../helpers';
import { useAxios, useClickOutside } from '../../hooks';
import { DEFAULT_BORDER_RADIUS, fontSize, pxToRem, spacing } from '../../styles';
import { TextField } from '../textField/textField';
import { Portal } from './portal';

const Wrapper = styled.div(({ themeMode }: WrapperProps) => {
    const { state: { theme } } = useStore();

    return css`
		display: flex;
		width: 100%;
		flex-direction: column;
		align-items: center;
        justify-content: space-around;
		color: ${themeMode === 'dark' ? '#000000' : themeMode === 'light' ? '#ffffff' : theme.font.default};
		`;
});

const Title = styled.h2(() => {

    return css`
	text-align: center;
	font-size: ${fontSize[18]};
	font-weight: normal;
    margin: 0;
	margin-bottom: ${spacing[20]};
`;
});

const InputContainer = styled.div(() => {
    return css`
        width: 100%;
        margin-bottom: ${spacing[10]};
    `;
});

const SubmitBtn = styled.button((props: any) => {

    return css`
		background-color: ${!props.disabled ? '#20A100' : 'grey'};
		width: 100%;
		max-width: ${pxToRem(200)};
		border-radius: ${DEFAULT_BORDER_RADIUS};
		border: none;
		padding: ${pxToRem(12)} 0;
		text-align: center;
		color: white;
		font-size: ${fontSize[16]};
        margin-bottom: ${spacing[20]};
		cursor: ${props.disabled ? 'not-allowed' : 'pointer'};
        transition: transform 0.7 ease;

        &:hover {
            transform: scale(1.06);
        }
	`;
});

type WrapperProps = {
    themeMode?: string;
};

type AdditionalAccount = {
    id: number;
    address: string;
    nonce: string;
    is_confirmed: boolean;
};

type Props = {
    showModal: boolean;
    setShowModal: (showModal: boolean, account?: AdditionalAccount) => void;
};

export const AddSubAccountModal = ({ showModal, setShowModal }: Props) => {
    const api = useAxios();
    const { state: { theme } } = useStore();
    const [address, setAddress] = useState<string>('');
    const isValid = address.length >= 2;

    const domNode = useClickOutside(() => {
        setShowModal(false);
    });

    const handleChangeInput = (event: any) => {
        setAddress(event.target.value);
    };

    const handleSubmit = async () => {
        try {
            const obj = {
                address,
            };
            await api.request({
                method: 'POST',
                url: `${BASE_URL}account/wallet/`,
                data: obj,
            }).then((res) => {
                setAddress('');
                const newAcc: AdditionalAccount = res.data;
                setShowModal(false, newAcc);
                toast.success('Additional account has been successfully added!', { theme: theme.name });
            });

        } catch (error) {
            setAddress('');
            toast.error('Something went wrong please try again!', { theme: theme.name });
            setShowModal(false);
            console.log('something went wrong please try again!', error);
        }
    };

    return (
        <Portal
            size='xs'
            isOpen={showModal}
            handleClose={() => setShowModal(false)}
            hasBackButton={false}
            backgroundColor='light'
            closeOutside
            themeMode='light'>
            {/* @ts-ignore */}
            <Wrapper ref={domNode} themeMode='dark'>
                <InputContainer>
                    <Title>Provide your additional account address</Title>
                    <TextField
                        size='small'
                        themeMode='light'
                        align='center'
                        onChange={handleChangeInput}
                        placeholder=" Additional account address"
                        type="text"
                        value={address}
                        maxLength={50}
                    />
                </InputContainer>
                <SubmitBtn
                    disabled={!isValid}
                    onClick={handleSubmit}>
                    Submit
                </SubmitBtn>
            </Wrapper>
        </Portal>
    );
};