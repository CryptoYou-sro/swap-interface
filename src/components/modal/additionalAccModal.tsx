import { useState } from 'react';
import styled, { css } from 'styled-components';
import { Portal } from '..';
import { useStore } from '../../helpers';
import { useMedia } from '../../hooks';
import { fontSize, mediaQuery, pxToRem, spacing } from '../../styles';

const ModalWrapper = styled.div`
	display: flex;
	align-items: center;
	justify-content: space-between;
	gap: ${spacing[8]};
	width: 100%;

	${mediaQuery('xs')} {
		flex-direction: column;
		align-items: flex-start;
		padding: ${spacing[18]} ${spacing[46]} 0 ${spacing[14]};
	}
`;

const AccountTitle = styled.div(() => {
    const {
        state: { theme }
    } = useStore();

    return css`
		font-size: ${fontSize[16]};
		color: ${theme.font.default};
		line-height: ${fontSize[22]};
		margin-bottom: ${spacing[20]};
	`;
});

const CopyContainer = styled.div`
	display: flex;
	align-items: center;
	cursor: pointer;

	${mediaQuery('xs')} {
		margin-bottom: ${spacing[16]};
	}
`;


const LinkContainer = styled.div(() => {
    const {
        state: { theme }
    } = useStore();

    return css`
		color: ${theme.font.default};
		margin-right: ${spacing[6]};
        
	`;
});

const IconContainer = styled.div(() => {
    const {
        state: { theme }
    } = useStore();

    return css`
		height: ${pxToRem(8)};
		width: ${pxToRem(10)};
		background-color: ${theme.button.transparent};
		border: 1px solid ${theme.font.default};
		transform: rotate(90deg);
		margin-right: ${spacing[8]};

		&::after {
			content: '';
			display: block;
			position: relative;
			top: -60%;
			right: -25%;
			height: ${pxToRem(8)};
			width: ${pxToRem(10)};
			border: 1px solid ${theme.font.default};
			background-color: ${theme.background.secondary};
		}
	`;
});

const Link = styled.a(() => {
    const {
        state: { theme }
    } = useStore();

    return css`
    &:visited {
		color: ${theme.font.default};
    }
    `;
});

const CopyText = styled.p.attrs((props: { isCopied: boolean }) => props)`
	color: ${({ isCopied }) => isCopied ? '#2ea8e8' : '#B4B4B4'};
	font-size: ${fontSize[14]};
	line-height: ${fontSize[20]};
	margin-top: ${pxToRem(18)};
`;

type Props = {
    showModal: boolean;
    setShowModal: (showModal: boolean) => void;
    link: string;
    isCopied?: boolean;
};

export const AdditionalAccModal = ({ showModal, setShowModal, link }: Props) => {
    const [isCopied, setIsCopied] = useState(false);
    const { mobileWidth: isMobile } = useMedia('xs');

    const handleCopy = () => {
        setIsCopied(true);
        void navigator.clipboard.writeText(link);
        setTimeout(() => {
            setIsCopied(false);
        }, 2000);
    };

    return (
        <Portal
            handleClose={() => setShowModal(false)}
            isOpen={showModal}
            size={isMobile ? 'small' : 'xs'}>
            <ModalWrapper>
                <div>
                    <AccountTitle>Special Link</AccountTitle>
                    <LinkContainer>
                        <Link href='https://app.cryptoyou.io/'>
                            {link}
                        </Link>
                    </LinkContainer>
                    <CopyContainer>
                        <IconContainer />
                        <CopyText onClick={handleCopy} isCopied={isCopied}>
                            {!isCopied ? 'Copy Link' : 'Copied!'}
                        </CopyText>
                    </CopyContainer>
                </div>
            </ModalWrapper>
        </Portal>
    );
};