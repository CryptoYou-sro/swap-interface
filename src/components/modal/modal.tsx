import { ReactNode, useEffect } from 'react';
import { useState } from 'react';
import styled, { css } from 'styled-components';
import { fontSize, mediaQuery, pxToRem, spacing } from '../../styles';
import { defaultBorderRadius, useStore, DestinationNetworkEnum } from '../../helpers';
import type { ThemeProps } from '../../styles';

const ModalWrapper = styled.div(({ width, showModal, background }: Props) => {
	const {
		state: { theme }
	} = useStore();

	return css`
		position: fixed;
		top: 50%;
		left: 50%;
		transform: translate(-50%, -50%);
		display: ${showModal ? 'block' : 'none'};
		justify-content: center;
		align-items: center;
		z-index: 100;
		width: ${pxToRem(width === 'large' ? 605 : 478)};
		max-width: calc(100% - ${spacing[64]});
		background-color: ${theme.background[background]};
		border: 1px solid ${theme.font.default};
		border-radius: ${defaultBorderRadius};
		padding: ${spacing[12]};

		${mediaQuery('xxs')} {
			width: calc(100% - ${pxToRem(10)});
		}
	`;
});

const CloseIcon = styled.div(({ theme }: ThemeProps) => {
	return css`
		cursor: pointer;
		position: fixed;
		top: ${pxToRem(10)};
		right: ${pxToRem(10)};
		font-size: ${fontSize[16]};
		line-height: ${fontSize[22]};
		color: ${theme.font.pure};
	`;
});

type Props = {
	showModal: boolean;
	setShowModal: (showModal: boolean) => void;
	width?: 'large' | 'small';
	background: 'default' | 'mobile';
	children?: ReactNode;
};

export const Modal = ({
	showModal,
	setShowModal,
	width = 'large',
	background,
	children
}: Props) => {
	const [selectedTokenNetwork, setSelectedTokenNetwork] = useState({ network: '', token: '' });
	const {
		state: { theme, destinationNetwork, destinationToken },
		dispatch
	} = useStore();

	useEffect(() => {
		if (showModal) {
			setSelectedTokenNetwork({ network: destinationNetwork, token: destinationToken });
		}
	}, [showModal]);

	const handleClose = () => {
		dispatch({ type: DestinationNetworkEnum.NETWORK, payload: selectedTokenNetwork.network });
		dispatch({ type: DestinationNetworkEnum.TOKEN, payload: selectedTokenNetwork.token });
		setShowModal(false);
	};

	return (
		<ModalWrapper
			width={width}
			showModal={showModal}
			background={background}
			data-testid="modal-container"
			setShowModal={setShowModal}
		>
			<CloseIcon onClick={handleClose} theme={theme}>
				&#x2716;
			</CloseIcon>
			{children}
		</ModalWrapper>
	);
};
