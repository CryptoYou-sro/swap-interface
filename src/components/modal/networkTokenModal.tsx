import _ from 'lodash';
import { useEffect, useMemo, useState } from 'react';
import styled from 'styled-components';
import { useDisconnect, useNetwork } from 'wagmi';
import { Button, Portal, SelectList, useToasts } from '../../components';
import {
	AmountEnum, CHAINS,
	DefaultSelectEnum,
	DestinationEnum, DestinationNetworks, NETWORK_TO_ID,
	SourceEnum,
	isNetworkSelected,
	isTokenSelected,
	useStore
} from '../../helpers';
import { useMedia } from '../../hooks';
import { mediaQuery, spacing } from '../../styles';

const Wrapper = styled.div`
	display: flex;
	width: 100%;
	justify-content: center;
	align-items: center;
	flex-direction: column;
	row-gap: ${spacing[22]};

	${mediaQuery('xs')} {
		flex-direction: column;
		flex-wrap: nowrap;
	}
`;

const SelectWrapper = styled.div`
	display: flex;
	flex-grow: 1;
	align-self: stretch;
	overflow: hidden;
	width: 100%;
	gap: ${spacing[18]};
`;

type Props = {
	showModal: boolean;
	setShowModal: (prev: boolean) => void;
	type: 'SOURCE' | 'DESTINATION';
};

export const NetworkTokenModal = ({ showModal, setShowModal, type }: Props) => {
	const { chain: wagmiChain } = useNetwork();
	const { disconnect } = useDisconnect();
	// @ts-ignore
	const { addToast } = useToasts();
	const [showsNetworkList, setShowsNetworkList] = useState(true);
	const { mobileWidth: isMobile } = useMedia('xs');
	const {
		dispatch,
		state: {
			destinationNetwork,
			destinationToken,
			sourceNetwork,
			sourceToken,
			availableSourceNetworks: SOURCE_NETWORKS,
			availableDestinationNetworks: DESTINATION_NETWORKS
		}
	} = useStore();

	const isSource = type === 'SOURCE';

	const isDisabled = useMemo(
		() =>
			isSource
				? !isNetworkSelected(sourceNetwork) || !isTokenSelected(sourceToken)
				: !isNetworkSelected(destinationNetwork) || !isTokenSelected(destinationToken),
		[destinationNetwork, destinationToken, sourceNetwork, sourceToken]
	);

	const sourceNetworksList = SOURCE_NETWORKS ? _.orderBy(
		Object.keys(SOURCE_NETWORKS).map(
			// @ts-ignore
			// eslint-disable-next-line
			(id) => CHAINS[id]?.name
		)
	) : [];

	const sourceTokensList = useMemo(
		() =>
			SOURCE_NETWORKS && isNetworkSelected(sourceNetwork)
				? // @ts-ignore
				_.orderBy(Object.keys(SOURCE_NETWORKS[NETWORK_TO_ID[sourceNetwork]]?.['tokens']))
				: [],
		[SOURCE_NETWORKS, sourceNetwork]
	);

	const destinationNetworksList = useMemo(() => {
		if (DESTINATION_NETWORKS && isNetworkSelected(sourceNetwork) && isTokenSelected(sourceToken)) {
			const allDestinationNetworks = Object.keys(
				// @ts-ignore
				DESTINATION_NETWORKS[NETWORK_TO_ID[sourceNetwork]]?.[sourceToken]
			);
			const filteredDestinationNetworks = allDestinationNetworks.filter((network) => {
				const tokens = Object.keys(
					// @ts-ignores
					DESTINATION_NETWORKS[NETWORK_TO_ID[sourceNetwork]]?.[sourceToken]?.[network]?.['tokens']
				);

				return tokens.join('') !== sourceToken;
			});

			return _.orderBy(filteredDestinationNetworks);
		} else {
			return [];
		}
	}, [DESTINATION_NETWORKS, sourceToken, sourceNetwork]);

	const destinationTokensList = useMemo(() => {
		if (DESTINATION_NETWORKS) {
			const tokensFromJson =
				// @ts-ignore
				DESTINATION_NETWORKS[NETWORK_TO_ID[sourceNetwork]]?.[sourceToken]?.[
				destinationNetwork as DestinationNetworks
				]?.['tokens'];

			const filteredTokens = (tokensFromJson ? Object.keys(tokensFromJson) : []).filter(
				(token) => token !== sourceToken
			);

			return _.orderBy(filteredTokens);
		} else {
			return [];
		}
	}, [DESTINATION_NETWORKS, sourceToken, destinationNetwork, sourceNetwork]);

	const handleSubmit = () => {
		setShowModal(!showModal);
	};

	const handleBack = () => {
		setShowsNetworkList(true);
		dispatch({
			type: isSource ? SourceEnum.TOKEN : DestinationEnum.TOKEN,
			payload: DefaultSelectEnum.TOKEN
		});
	};

	useEffect(() => {
		setShowsNetworkList(true);
	}, [showModal]);

	useEffect(() => {
		console.log(wagmiChain);

		if (wagmiChain && Object.keys(CHAINS).includes(wagmiChain?.id.toString())) {
			dispatch({
				type: SourceEnum.NETWORK,
				// @ts-ignore
				payload: CHAINS[wagmiChain?.id.toString()]?.name
			});
			dispatch({
				type: SourceEnum.TOKEN,
				// @ts-ignore
				payload: CHAINS[wagmiChain?.id.toString()].name
			});
			dispatch({ type: AmountEnum.AMOUNT, payload: '' });
			dispatch({ type: DestinationEnum.AMOUNT, payload: '' });
		} else if (wagmiChain && !Object.keys(CHAINS).includes(wagmiChain?.id.toString())) {
			disconnect();
			addToast('Please change the network to one that is supported', 'warning');
		}
	}, [wagmiChain]);

	return !isMobile ? (
		<Portal handleClose={() => setShowModal(false)} isOpen={showModal} size="large">
			{(isSource ? sourceNetworksList : destinationNetworksList)?.length > 0 && (
				<Wrapper>
					<SelectWrapper>
						<SelectList
							value={isSource ? 'SOURCE_NETWORK' : 'NETWORK'}
							data={isSource ? sourceNetworksList : destinationNetworksList}
							placeholder="Network Name"
						/>
						<SelectList
							value={isSource ? 'SOURCE_TOKEN' : 'TOKEN'}
							data={isSource ? sourceTokensList : destinationTokensList}
							placeholder="Token Name"
						/>
					</SelectWrapper>
					<Button disabled={isDisabled} onClick={handleSubmit} color="transparent">
						{isDisabled ? 'Select Network and Token' : 'Select'}
					</Button>
				</Wrapper>
			)}
		</Portal>
	) : (
		<Portal
			size="large"
			handleClose={() => setShowModal(false)}
			isOpen={showModal}
			hasBackButton={!showsNetworkList}
			handleBack={handleBack}>
			<Wrapper>
				{/* @ts-ignore */}
				{(isSource ? sourceNetworksList : destinationNetworksList)?.length > 0 ? (
					<>
						{showsNetworkList && (
							<SelectList
								value={isSource ? 'SOURCE_NETWORK' : 'NETWORK'}
								data={isSource ? sourceNetworksList : destinationNetworksList}
								placeholder="Network Name"
							/>
						)}
						{!showsNetworkList && (
							<SelectList
								value={isSource ? 'SOURCE_TOKEN' : 'TOKEN'}
								data={isSource ? sourceTokensList : destinationTokensList}
								placeholder="Token Name"
							/>
						)}
					</>
				) : (
					<div>No available networks...</div>
				)}
				{showsNetworkList && (
					<Button
						color="transparent"
						onClick={() => setShowsNetworkList(false)}
						disabled={!isNetworkSelected(isSource ? sourceNetwork : destinationNetwork)}>
						Next
					</Button>
				)}
				{!showsNetworkList && (
					<Button color="transparent" onClick={handleSubmit} disabled={isDisabled}>
						{isDisabled ? 'Select Token' : 'Select'}
					</Button>
				)}
			</Wrapper>
		</Portal>
	);
};
