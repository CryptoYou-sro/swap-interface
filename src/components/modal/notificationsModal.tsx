import { useState } from 'react';
import styled from 'styled-components';
import { Portal, TextField, Button, Switch } from '../../components';
import { spacing } from '../../styles';

const Wrapper = styled.div`
	display: flex;
	justify-content: center;
	flex-direction: column;
	gap: ${spacing[28]};
	width: 100%;
`;

const Newsletter = styled.div`
	display: flex;
	justify-content: space-between;
	gap: ${spacing[8]};

	& input {
		flex: 2;
	}

	& button {
		flex: 1;
	}
`;

const Notifications = styled.div`
	display: flex;
	justify-content: flex-start;
	align-items: center;
	gap: ${spacing[12]};
`;

type Props = {
	showModal: boolean;
	setShowModal: (showModal: boolean) => void;
};

export const NotificationsModal = ({ showModal, setShowModal }: Props) => {
	const [email, setEmail] = useState('');

	return (
		<Portal handleClose={() => setShowModal(false)} isOpen={showModal}>
			<Wrapper>
				<Newsletter>
					<TextField
						value={email}
						size="small"
						placeholder="Insert email address"
						onChange={(e) => setEmail(e.target.value)}
					/>
					<Button variant="secondary" color="default" onClick={() => console.log('save email')}>
						Save
					</Button>
				</Newsletter>
				<Notifications>
					Notifications
					<Switch />
				</Notifications>
			</Wrapper>
		</Portal>
	);
};
