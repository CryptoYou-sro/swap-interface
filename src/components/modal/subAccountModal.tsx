import { useState } from 'react';
import styled, { css } from 'styled-components';
import { useDisconnect } from 'wagmi';
import { Portal } from '..';
import { useStore } from '../../helpers';
import { useClickOutside, useMedia } from '../../hooks';
import { fontSize, mediaQuery, pxToRem, spacing } from '../../styles';

const ModalWrapper = styled.div(({ themeMode }: any) => {
    const {
        state: { theme }
    } = useStore();

    return css`
    width: 100%;
    color: ${themeMode === 'dark' ? '#000000' : themeMode === 'light' ? '#ffffff' : theme.font.default};
`;
});

const AccountWrapper = styled.div(() => {
    return css`
        display: flex;
        align-items: center;
        justify-content: space-between;
        flex-direction: column;        
    `;
});

const AccountTitle = styled.h1(() => {
    return css`
        font-size: ${fontSize[18]};
        line-height: ${fontSize[22]};
        margin: 0;
    `;
});

const CopyContainer = styled.div`
    display: flex;
    align-items: center;
    cursor: pointer;
    width: 100%;
    justify-content: flex-end;

    ${mediaQuery('xs')} {
        margin-bottom: ${spacing[16]};
    }
`;


const LinkContainer = styled.div(() => {
    const { mobileWidth: isMobile } = useMedia('s');

    return css`
        margin: 0;
        margin-right: ${spacing[6]};
        margin-bottom: ${spacing[10]};
        word-break: break-all;
        width: 100%;
        max-width: ${isMobile ? pxToRem(260) : pxToRem(450)};
    `;
});

const IconContainer = styled.div(() => {
    const {
        state: { theme }
    } = useStore();

    return css`
        height: ${pxToRem(8)};
        width: ${pxToRem(10)};
        background-color: ${theme.button.transparent};
        border: 1px solid black;
        transform: rotate(90deg);
        margin-right: ${spacing[8]};

        &::after {
            content: '';
            display: block;
            position: relative;
            top: -60%;
            right: -25%;
            height: ${pxToRem(8)};
            width: ${pxToRem(10)};
            border: 1px solid black;
            background-color: white;
        }
    `;
});

const TextLink = styled.p(() => {
    return css`
        cursor: pointer;
        text-align: left;
        margin-bottom: 0;

        &:hover {
            text-decoration: underline;
        }
    `;
});

const CopyText = styled.p.attrs((props: { isCopied: boolean }) => props)`
    color: ${({ isCopied }) => isCopied ? '#2ea8e8' : '#000000'};
    font-size: ${fontSize[14]};
    line-height: ${fontSize[20]};
    margin-top: ${pxToRem(18)};
`;

const ConfirmedText = styled.p(() => {
    return css`
    font-size: ${fontSize[18]};
    `;
});

type AdditionalAccount = {
    id: number;
    address: string;
    nonce: string;
    is_confirmed: boolean;
};

type Props = {
    showModal: boolean;
    setShowModal: (showModal: boolean) => void;
    account: AdditionalAccount;
    isCopied?: boolean;
};

export const SubAccountModal = ({ showModal, setShowModal, account }: Props) => {
    const { disconnect } = useDisconnect();
    const [isCopied, setIsCopied] = useState(false);
    const isConfirmed = account.is_confirmed;

    const handleCopy = () => {
        setIsCopied(true);
        void navigator.clipboard.writeText(`https://app.cryptoyou.io/?address=${account.address}&nonce=${account.nonce}`);
        setTimeout(() => {
            setIsCopied(false);
        }, 2000);
    };

    const domNode = useClickOutside(() => {
        setShowModal(false);
    });

    const handleLink = () => {
        disconnect();
        window.location.href = `https://app.cryptoyou.io/?address=${account.address}&nonce=${account.nonce}`;
    };


    return (
        <Portal
            size='small'
            isOpen={showModal}
            handleClose={() => setShowModal(false)}
            hasBackButton={false}
            backgroundColor='light'
            themeMode='light'>
            {/* @ts-ignore */}
            <ModalWrapper ref={domNode} themeMode='dark'>
                {!isConfirmed ? (
                    <AccountWrapper>
                        <AccountTitle>Verification link</AccountTitle>
                        <p>To verify your additional account, follow the link below and log in to your additional account.</p>
                        <LinkContainer>
                            <TextLink onClick={handleLink}>
                                https://app.cryptoyou.io/?address=${account.address}&nonce=${account.nonce}
                            </TextLink>
                        </LinkContainer>
                        <CopyContainer>
                            <IconContainer />
                            <CopyText onClick={handleCopy} isCopied={isCopied}>
                                {!isCopied ? 'Copy Link' : 'Copied!'}
                            </CopyText>
                        </CopyContainer>
                    </AccountWrapper>
                ) : (
                    <ConfirmedText>
                        Account
                        <span style={{ fontWeight: 'bold' }}> {account.address.slice(0, 10)}...{account.address.slice(-5)} </span>
                        already confirmed!
                    </ConfirmedText>
                )}
            </ModalWrapper>
        </Portal>
    );
};