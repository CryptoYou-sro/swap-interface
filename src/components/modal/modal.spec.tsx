import 'jest-styled-components';
import { render } from '@testing-library/react';
import { AuthProvider } from '../../helpers';
import { Modal } from './modal';
import {
	darkTheme,
	DEFAULT_BORDER_RADIUS,
	pxToRem,
	SELECT_LIST_HEIGHT,
	spacing
} from '../../styles';

describe('Modal', () => {
	it('should render a modal with background mobile and large size', () => {
		const { getByTestId, getByText } = render(
			<AuthProvider>
				<Modal showModal setShowModal={() => console.log()} background="mobile" width="large" />
			</AuthProvider>
		);

		const modal = getByTestId('modal-container');
		expect(modal).toMatchSnapshot();
		expect(modal).toBeInTheDocument();
		expect(modal).toHaveStyle(`
			position: fixed;
			top: 50%;
			left: 50%;
			transform: translate(-50%,-50%);
			display: block;
			justify-content: center;
			align-items: center;
			z-index: 100;
			width: ${pxToRem(605)};
			max-width: calc(100% - ${spacing[64]});
			background-color: ${darkTheme.background.mobile};
			border: 1px solid ${darkTheme.font.default};
			border-radius: ${DEFAULT_BORDER_RADIUS};
			padding: ${spacing[12]};
			`);

		const mobileMedia = '(max-width:375px)';

		expect(modal).toHaveStyleRule('width', `calc(100% - ${pxToRem(10)})`, {
			media: mobileMedia
		});

		const closeIcon = getByText('âœ–');
		expect(closeIcon).toBeInTheDocument();
		expect(closeIcon).toHaveStyle(`
			cursor: pointer;
			position: fixed;
			top: ${pxToRem(10)};
			right: ${pxToRem(10)};
			font-size: ${pxToRem(16)};
			line-height: ${pxToRem(22)};
			color: ${darkTheme.font.pure};
			`);
	});

	it('should render modal with small size and default background', () => {
		const { getByTestId } = render(
			<AuthProvider>
				<Modal showModal setShowModal={() => console.log()} background="default" width="small" />
			</AuthProvider>
		);

		const modal = getByTestId('modal-container');
		expect(modal).toHaveStyle(`
			width: ${SELECT_LIST_HEIGHT};
			background-color: ${darkTheme.background.default};
			`);
	});
});
