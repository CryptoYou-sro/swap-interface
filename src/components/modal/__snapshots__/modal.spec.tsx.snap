// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render a modal with background mobile and large size 1`] = `
.c0 {
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  display: block;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 100;
  width: 37.8125rem;
  max-width: calc(100% - 4rem);
  background-color: #1C2125;
  border: 1px solid #8C8D8F;
  border-radius: 0.375rem;
  padding: 0.75rem;
}

.c1 {
  cursor: pointer;
  position: fixed;
  top: 0.625rem;
  right: 0.625rem;
  font-size: 1rem;
  line-height: 1.375rem;
  color: #FFF;
}

@media (max-width:375px) {
  .c0 {
    width: calc(100% - 0.625rem);
  }
}

<div
  class="c0"
  data-testid="modal-container"
  width="large"
>
  <div
    class="c1"
  >
    âœ–
  </div>
</div>
`;
